sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -163,7 +163,14 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
     val table = client.getTable(databaseName, tblName)
 
     if (table.getProperty("spark.sql.sources.provider") != null) {
-      cachedDataSourceTables(QualifiedTableName(databaseName, tblName).toLowerCase)
+      val datasourceTable =
+        cachedDataSourceTables(QualifiedTableName(databaseName, tblName).toLowerCase)
+      // Wrap the table with a Subquery with the table name.
+      val tableWithQualifers = Subquery(tableIdent.last, datasourceTable)
+      // Then, if alias is specified, wrap the table again with the alias in a Subquery.
+      val withAlias = alias.map(a => Subquery(a, tableWithQualifers)).getOrElse(tableWithQualifers)
```

Comment:
Nit: the double subquery here does nothing.  I'd just use `dataSourceTable` in the `.map(...)`